packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -39,79 +46,91 @@ export default function NavigationPlaceholder( {
-		hasMenus,
-	} = useNavigationEntities();
-
-	const isStillLoading = isResolvingPages || isResolvingMenus;
+	const { hasMenus, isResolvingMenus } = useNavigationEntities();
 
 	const onCreateEmptyMenu = () => {
 		onFinishMenuCreation( [] );
 	};
 
 	const { navigationMenus } = useNavigationMenu();
 
+	useEffect( () => {
+		if ( isResolvingMenus ) {
+			speak(
+				'Loading Navigation block setup placeholder options.',
+				'polite'
+			);
+		} else {
+			speak(
+				'Navigation block setup placeholder options ready.',
+				'polite'
+			);
+		}
+	}, [ isResolvingMenus ] );
+
 	const hasNavigationMenus = !! navigationMenus?.length;
 
 	const showSelectMenus =
 		( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
 		( hasNavigationMenus || hasMenus );
 
+	const isResolvingActions =
+		isResolvingMenus && isResolvingCanUserCreateNavigationMenu;
+
 	return (
 		<>
-			{ ( ! hasResolvedNavigationMenus || isStillLoading ) && (
-				<PlaceholderPreview isLoading />
-			) }
-			{ hasResolvedNavigationMenus && ! isStillLoading && (
-				<Placeholder className="wp-block-navigation-placeholder">
-					<PlaceholderPreview />
-					<div className="wp-block-navigation-placeholder__controls">
-						<div className="wp-block-navigation-placeholder__actions">
-							<div className="wp-block-navigation-placeholder__actions__indicator">
-								<Icon icon={ navigation } />{ ' ' }
-								{ __( 'Navigation' ) }
-							</div>
-
-							<hr />
-
-							{ showSelectMenus ? (
-								<>
-									<DropdownMenu
-										text={ __( 'Select menu' ) }
-										icon={ null }
-										toggleProps={ {
-											variant: 'tertiary',
-											iconPosition: 'right',
-											className:
-												'wp-block-navigation-placeholder__actions__dropdown',
-										} }
-										popoverProps={ { isAlternate: true } }
-									>
-										{ () => (
-											<NavigationMenuSelector
-												clientId={ clientId }
-												onSelect={ onFinish }
-											/>
-										) }
-									</DropdownMenu>
-									<hr />
-								</>
-							) : undefined }
-
-							{ canUserCreateNavigationMenu && (
-								<Button
-									variant="tertiary"
-									onClick={ onCreateEmptyMenu }
-								>
-									{ __( 'Start empty' ) }
-								</Button>
-							) }
+			<Placeholder className="wp-block-navigation-placeholder">
+				{
+					// The <PlaceholderPreview> component is displayed conditionally via CSS depending on
+					// whether the block is selected or not. This is achieved via CSS to avoid
+					// component re-renders
+				 }
+				<PlaceholderPreview />
+				<div className="wp-block-navigation-placeholder__controls">
+					<div className="wp-block-navigation-placeholder__actions">
+						<div className="wp-block-navigation-placeholder__actions__indicator">
+							<Icon icon={ navigation } /> { __( 'Navigation' ) }
 						</div>
+
+						<hr />
+
+						{ isResolvingActions && <Spinner /> }
+
+						{ showSelectMenus ? (
```

Comment:
Now that we've removed the "Add all pages" option, does it even make sense to show the placeholder if there are no existing menus to select? It feels a bit weird to have a placeholder with a single option inside it:

<img width="871" alt="Screen Shot 2022-02-21 at 5 27 34 pm" src="https://user-images.githubusercontent.com/8096000/154900729-aa4d32a3-74cd-4792-bc92-8f43adaa13ea.png">

Reply:
This is an excellent and pertinent question touching on a larger issue that'd be good to explore. Overall, it would be very nice if we could rid of the placeholder in most cases, such as if you already have a single menu defined, then activate a new theme, just load that first menu automatically, since it's so easy to swap it out from the toolbar anyway.

The other is the behavior of when _zero_ menus are defined. If I understand this correctly, the "Page List" block is currently output automatically on frontends in empty menus, is that right? While useful in principle it does make for a bit of a disconnect between frontend and editor, that could result in confusion: "I'm seeing all my pages here, but not there". In that light, I wonder if we should just preinsert the Page List block in navigation, if no saved menus are created, so the two are in sync? I'm assuming no, based on past discussions on navigation persistence, but I'm suggesting the concept in case it can inspire other ideas. We still don't really have a good way to create patterns for the navigation either.

The new empty state is inspired by Site Logo and Featured Image, and features this dashed line:

<img width="683" alt="Screenshot 2022-02-21 at 10 26 48" src="https://user-images.githubusercontent.com/1204802/154926491-8042a370-eb0b-4e20-88c7-16d08062f24a.png">

Outside of somehow pre-populating an empty block, it seems prudent to at least keep that state around to indicate: there's an empty navigation block here which you can click to configure. If we can have that state, and go directly to the nav inserter on select, in principle that sounds good. But wouldn't that require us to defer the saving to a CPT? Otherwise simply clicking the dashed line nav block would presumable create an empty menu which was then loaded by default? 🤔